aardwolf-models/examples/local_user_auth.rs

```diff
@@ -0,0 +1,210 @@
+                    .unwrap(),
+                &connection,
+            );
+
+            let (new_email, token) = NewEmail::new(payload.email, &user).unwrap();
+
+            insert_email(new_email, &connection);
+
+            println!("Created user, local_auth, and email!");
+            token
+        };
+
+        // Log in the unverified user
+        {
+            let json = json!({
+                "email": "test@example.com",
+                "password": "testpass",
+            });
+
+            let payload: AuthPayload = serde_json::from_value(json).unwrap();
+
+            let (unauthenticated_user, _email, local_auth) =
+                lookup_user_by_email(payload.email, &connection);
+
+            let user = unauthenticated_user
+                .log_in_local(local_auth, payload.password)
+                .unwrap();
+
+            assert!(
+                !user.is_verified(&connection).unwrap(),
+                "User shouldn't be verified at this point"
+            );
+
+            println!("Logged in unverified User!!!");
+        }
+
+        // Verify the user
+        {
+            let json = json!({
+                "email":"test@example.com",
+                "token":format!("{}", token)
+            });
+
+            let payload: VerificationPayload = serde_json::from_value(json).unwrap();
+
+            let (unauthenticated_user, email, _local_auth) =
+                lookup_user_by_email(payload.email, &connection);
+
+            let unverified_user = unauthenticated_user
+                .to_verified(&connection)
+                .unwrap()
+                .unwrap_err();
+
+            let unverified_email = match email.to_verified() {
+                Ok(_) => panic!("Unexpected verified email"),
+                Err(unverified_email) => unverified_email,
+            };
+
+            let (_authenticated_user, _verified_email) = unverified_user
+                .verify(unverified_email, payload.token)
+                .unwrap()
+                .store_verify(&connection)
+                .unwrap();
+
+            println!("Verified user!");
+        }
+
+        // log in the verified user
+        {
+            let json = json!({
+                "email": "test@example.com",
+                "password": "testpass",
+            });
+
+            let payload: AuthPayload = serde_json::from_value(json).unwrap();
+
+            let (unauthenticated_user, _email, local_auth) =
+                lookup_user_by_email(payload.email, &connection);
+
+            let user = unauthenticated_user
+                .log_in_local(local_auth, payload.password)
+                .unwrap();
+
+            assert!(
+                user.is_verified(&connection).unwrap(),
+                "User should be verified at this point"
+            );
+
+            println!("Logged in verified User!!!");
+        }
+
+        Ok(())
+    });
+
+    println!("Hewwo, Mr Obama???");
```

Comment:
Misspelling on 209

Should be "Thanks Obama!" 

;D

Reply:
>:c

Reply:
actually though, I need to update that example

Reply:
*pats* its a joke fren